Serverless guide improvements (#9267)
dagster-io/dagster: docs/content/dagster-cloud/deployment/serverless.mdx

## Getting started with Serverless

> 

Reviewer: In other folks' guides, I've had them add a list of subsections with links in cases like this. It lets the user see what's in the section while remaining inline, and also allows them to easily jump to the next section.

```suggestion

- [With GitHub](#with-github)
- [Without GitHub (GitLab, BitBucket, or local development](#without-github-gitlab-bitbucket-or-local-development)
- [Adding secrets](#adding-secrets)

```

### With GitHub

If you are a GitHub user, our GitHub integration is the fastest way to get started. It uses a GitHub app and GitHub Actions to set up a repo containing skeleton code and configuration consistent with Dagster Cloud's best practices with a single click.

> When you create a new Dagster Cloud organization, you'll be prompted to choose Serverless or Hybrid. Once activated, our GitHub integration will scaffold a new git repo for you with serverless and branch deployments preconfigured. Pushing to the `main` branch will deploy to your `prod` Serverless deployment. Pull requests will spin up ephemeral [branch deployments](/dagster-cloud/developing-testing/branch-deployments) using the Serverless agent.

Reviewer: When referring to proprietary features, we're considering them to be proper nouns and capitalizing them - so `Serverless` and `Branch Deployments`

When referring to them in a generic context - ex: `a branch deployment`, we should use lower case

```suggestion
When you create a new Dagster Cloud organization, you'll be prompted to choose Serverless or Hybrid deployment. Once activated, our GitHub integration will scaffold a new git repo for you with Serverless and Branch Deployments already configured. Pushing to the `main` branch will deploy to your `prod` Serverless deployment. Pull requests will spin up ephemeral [branch deployments](/dagster-cloud/developing-testing/branch-deployments) using the Serverless agent.
```

Reviewer: Additionally, is this correct? I thought this was true if the user opts to clone the starter repo, not if they use an existing one.

> Once activated, our GitHub integration will scaffold a new git repo for you

### Without GitHub (GitLab, BitBucket, or local development)

If you don't want to use our GitHub integration, we offer a powerful CLI that you can use in another CI environment or on your local laptop.

> First, let's [create a new project](https://docs.dagster.io/getting-started/create-new-project#create-a-new-project) with the Dagster open-source CLI.

Reviewer: ```suggestion
First, [create a new project](https://docs.dagster.io/getting-started/create-new-project#create-a-new-project) with the Dagster open-source CLI.
```

Suggesting we remove `let's` here. The rest of the instructions in this section use an implied subject/understood you, so making this one consistent feels less awkward to me.

```shell
pip install dagster
dagster project scaffold \
  --name my-dagster-project \
  --example assets_modern_data_stack
```

Once scaffolded, add `dagster-cloud` as a dependency in your `setup.py` file.

Next, install the [dagster-cloud CLI](/dagster-cloud/developing-testing/dagster-cloud-cli) and log in to your org. **Note**: The CLI requires a recent version of Python 3 and Docker.
```shell
pip install dagster-cloud
dagster-cloud configure
```

> Finally, deploy your project with Dagster Cloud Serverless

Reviewer: ```suggestion
Finally, deploy your project with Dagster Cloud Serverless:
```

Reviewer: Adding a colon to tie the sentence to the example + add ending punctuation.
